glommio/src/io/directory.rs

```diff
@@ -87,6 +88,25 @@ impl Directory {
         Self::sync_open(&path)
     }
 
+    /// Creates a file under this directory, returns a DMA file
+    ///
+    /// NOTE: Path must not contain directories and just be a file name
+    pub async fn create_file<P: AsRef<Path>>(&self, path: P) -> io::Result<DmaFile> {
+        if path
+            .as_ref()
+            .to_string_lossy()
```

Comment:
It is not a good idea to create a `String` every time this is called.
If the argument is replaced with `AsRef<OsStr>`, `Path:new` would be called here, and I think it would be a good idea to call `.components().count() > 1`.

Reply:
We could however create a `PathBuf`. We store the `PathBuf` in the `GlommioFile`, so we would be just creating it earlier. `open_at` would have to change so that it could take a `PathBuf` as well, but that's doable.

Reply:
Ok, I looked at the actual patch now and I agree with @zserik .

The best way to validate this is with `.components()` as path without transforming it into anything: nor a `PathBuf` nor a `String`

I am kind of in the fence with `OsStr`. `Path` seems more inline with the rest of the File.

Reply:
yeah makes sense, i wasn't sure if path `/` should be valid or not.

Thanks looking!